src/coordinates/uniform_cylindrical.hpp

```diff
@@ -0,0 +1,517 @@
+  // Generalized physical volume
+  //----------------------------------------
+  template <TopologicalElement el, class... Args>
+  KOKKOS_FORCEINLINE_FUNCTION Real Volume(Args... args) const {
+    using TE = TopologicalElement;
+    if constexpr (el == TE::CC) {
+      return CellVolume(args...);
+    } else if constexpr (el == TE::F1) {
+      return FaceArea<X1DIR>(args...);
+    } else if constexpr (el == TE::F2) {
+      return FaceArea<X2DIR>(args...);
+    } else if constexpr (el == TE::F3) {
+      return FaceArea<X3DIR>(args...);
+    } else if constexpr (el == TE::E1) {
+      return EdgeLength<X1DIR>(args...);
+    } else if constexpr (el == TE::E2) {
+      return EdgeLength<X2DIR>(args...);
+    } else if constexpr (el == TE::E3) {
+      return EdgeLength<X3DIR>(args...);
+    } else if constexpr (el == TE::NN) {
+      return 1.0;
+    }
+    PARTHENON_FAIL("If you reach this point, someone has added a new value to the the "
+                   "TopologicalElement enum.");
+    return 0.0;
+  }
+
+  //----------------------------------------
+  // Cylindrical+Spherical Conversions
+  //----------------------------------------
+  // TODO
+
+  //----------------------------------------
+  // Geometric Terms (Find better names for these!)
+  //----------------------------------------
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real h2v(const int i) const { return r_c_(i); };
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real h2f(const int i) const { return Xf<1>(i); };
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real h31v(const int i) const { return 1.0; };
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real h31f(const int i) const { return 1.0; };
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real dh2vd1(const int i) const { return 1.0; };
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real dh2fd1(const int i) const { return 1.0; };
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real dh31vd1(const int i) const { return 0.0; };
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real dh31fd1(const int i) const { return 0.0; };
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real h32v(const int j) const { return 1.0; }
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real h32f(const int j) const { return 1.0; }
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real dh32vd2(const int j) const { return 0.0; }
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real dh32fd2(const int j) const { return 0.0; }
+
+  //What is this?
+  KOKKOS_FORCEINLINE_FUNCTION
+  void GetAcc(const int k, const int j, const int i, const Real &rp,
+	      const Real &cosphip, const Real &sinphip, const Real &zp,
+	      Real &acc1, Real &acc2, Real &acc3) const
+  {
+    const Real cosdphi = cosphi_c_(j)*cosphip - sinphi_c_(j)*sinphip; //cos(x3v(k)-phip)
+    const Real sindphi = sinphi_c_(j)*cosphip - cosphi_c_(j)*sinphip; //sin(x3v(k)-phip)
+    //Psi = - 1.0/sqrt(r0^2 + rp^2 - 2r0*rp*cos(phi-phip) + (z-zp)^2)
+    acc1 = (r_c_(i) - rp*cosdphi); //acc1 = dPsi/dr, Psi=-1/dist2
+    acc2 = rp*sindphi;        //acc2 = dPsi/(r*dphi) 
+    acc3 = Xc<3>(k)-zp;   
+  }
```

Comment:
this looks to me like a volume-integrated source term.

Reply:
Actually after grepping through Athena++ I think `GetAcc` is completely unused. I would recommend we don't include it in parthenon.

Reply:
I just checked Shengtai's code, it's used there in a commented out line of a problem generator specific to that code. So it should be left out of Parthenon